src/coreComponents/constitutive/solid/DelftEgg.hpp

```diff
@@ -0,0 +1,618 @@
+  real64 a = alpha;
+  real64 pa = pc;
+  real64 factor = 1.0;
+
+  if( p >= c ) // Use MCC
+  {
+    a = 1.0;
+    factor = 2.*alpha/ (alpha+1.);
+    pa = factor * pc;
+  }
+  real64 alphaTerm = 2. * a*a*a / (a+1.);
+  df_dp = -alphaTerm * pa + 2. * a * a * p;
+  df_dq = 2. * q /(M*M);
+  df_dpc = 2. * a*a*(a-1.) /(a+1.) * pc - alphaTerm * p * factor;
+  real64 dpc_dve = -1./(Cc-Cr) * pc;
+  df_dp_dve = 2. * a * a * bulkModulus + alphaTerm * dpc_dve * factor;
+  df_dq_dse = 2. /(M*M) * 3. * mu;
+
+  f = q*q/(M*M)- a*a*p *(2.*a/(a+1.)*pa-p)+a*a*(a-1.)/(a+1.)* pc*pc;
+
+}
+
+
+GEOSX_HOST_DEVICE
+GEOSX_FORCE_INLINE
+void DelftEggUpdates::smallStrainUpdate( localIndex const k,
+                                         localIndex const q,
+                                         real64 const ( &strainIncrement )[6],
+                                         real64 ( & stress )[6],
+                                         real64 ( & stiffness )[6][6] ) const
+{
+
+  // Rename variables for easier implementation
+
+  real64 const oldPc  = m_oldPreConsolidationPressure[k][q];   //pre-consolidation pressure
+  real64 const mu     = m_shearModulus[k];
+  real64 const bulkModulus     = m_bulkModulus[k];
+
+  real64 const M      = m_cslSlope[k];
+  real64 const Cr     = m_recompressionIndex[k];
+  real64 const Cc     = m_virginCompressionIndex[k];
+  real64 const alpha  = m_shapeParameter[k];
+
+  real64 pc    = oldPc;
+
+  // elastic predictor (assume strainIncrement is all elastic)
+
+  ElasticIsotropicUpdates::smallStrainUpdate( k, q, strainIncrement, stress, stiffness );
+
+  real64 trialP;
+  real64 trialQ;
+  real64 eps_v_trial;
+  real64 eps_s_trial;
+  real64 deviator[6];
+
+  twoInvariant::stressDecomposition( stress,
+                                     trialP,
+                                     trialQ,
+                                     deviator );
+
+  // check yield function F <= 0
+
+  real64 yield, df_dp, df_dq, df_dpc, df_dp_dve, df_dq_dse;
+  evaluateYield( trialP, trialQ, pc, M, alpha, Cc, Cr, bulkModulus, mu, yield, df_dp, df_dq, df_dpc, df_dp_dve, df_dq_dse );
+
+
+  if( yield < 1e-9 ) // elasticity
+  {
+    return;
+  }
+
+// else, plasticity (trial stress point lies outside yield surface)
+
+  eps_v_trial = trialP/bulkModulus;
+
+  eps_s_trial = trialQ/3.0/mu;
+
+  real64 solution[3], residual[3], delta[3];
+  real64 jacobian[3][3] = {{}}, jacobianInv[3][3] = {{}};
+
+  solution[0] = eps_v_trial; // initial guess for elastic volumetric strain
+  solution[1] = eps_s_trial; // initial guess for elastic deviatoric strain
+  solution[2] = 1e-10;     // initial guess for plastic multiplier
+
+  real64 norm, normZero = 1e30;
+  integer cuts = 0;
+  integer maxCuts = 100;
+  real64 normOld = normZero;
+
+  // begin Newton loop
+
+  for( localIndex iter=0; iter<20; ++iter )
+  {
+
+    trialP = solution[0] * bulkModulus;
+
+    trialQ = 3. * mu * solution[1];
+
+    // real64 h = 1.0 / (Cc-Cr); //Linear hardening version
```

Comment:
delete unused code that is not worth saving

Reply:
done.